docs/application/web/guides/applications/addon.md

```diff
@@ -0,0 +1,200 @@
+
+The main JavaScript file must have the same name with `main` in the manifest file. The main JavaScript file includes:
+
+  - `activate()`: To register a callback to an addon event.
+  - `deactivate()`: To unregister the callback to the addon event.
+  - `Callback Functions`: You can write your own code to define the behavior that happens when an addon event occurs.
+
+Following is an example of a main JavaScript file:
+
+  ```
+  class AddonSampleAd {
+      constructor(em) {
+        console.log('AddonSampleAd constructor');
+        this.evt = em;
+      }
+      activate() {
+          console.log('AddonSampleAd.activate');
+          this.evt.on('lcPrelaunch', this.prelaunch);
+      }
+      deactivate() {
+          console.log('AddonSampleAd.dectivate');
+          this.evt.off('lcPrelaunch', this.prelaunch);
+      }
+      prelaunch(winId, orgUrl) {
+          console.log('AddonSampleAd.prelaunch');
+      }
+  }
+  module.exports = AddonSampleAd;
+  ```
+
+<a name="event"></a>
+## Event
+
+| Event                                      | Description                              |
+| ------------------------------------------ | ---------------------------------------- |
+| `lcPrelaunch` | Emitted once for Web app launching just before the main page is shown. |
+| `lcResume` | Emitted when the app is resumed. |
+| `lcSuspend` | Emitted when the app is suspended. |
+| `lcQuit` | Emitted before the app is ended. |
+| `contentDidFinishLoad` | Emitted when the page navigation is done. | 
+
+<a name="mechanism"></a>
+## Mechanism
+
+Addons are not standalone software programs, and only run when a Web application runs. They work on event-driven invocation. Each addon registers its callbacks to events that are pre-defined in Web Runtime JS (WRTjs). Registered callbacks are invoked whenever the registered events occur during the Web application’s life cycle.  
+
+**Figure: Addon Mechanism**
+
+![Addon mechanism](./media/addon_mechanism.png)
+
+<a name="compare"></a>
+## Comparing to Chrome Extensions
+
+The following are the differences from Chrome Extensions:
+
+1. Applying to Multiple Applications
+
+   Chrome Extensions are applied to multiple tabs in an application named Chrome browser. However, addons are applied to multiple Web applications in a platform such as Tizen.
+
+2. Wide Range of Customization
+
+   Addons can customize an action when each Web application’s life cycle event such as Prelaunch, Resume, Suspend, or Quit occurs. Addons get the notification after they register its own callback for a lifecycle event. That is possible because they run on WRTjs, also known as Web Runtime JS, which manages and controls from birth to death of Web applications.
+
+<a name="api"></a>
+## API
+
+Addon APIs are provided to support the way to handle the window in both browser process and renderer process. More APIs will be added in the future. The renderer process here is only available when the addon itself has HTML files and they are loaded.
```

Comment:
Please remove article "the" before future:

More APIs will be added in future.

Reply:
done